### PR TITLE
DO_NOT_MERGE add invalidx to authorizationdao builder

### PR DESCRIPTION
**Pull request by noodlemoodle**
https://github.com/nanoPayinc/NANOPAY/pull/8113 nanopay pr

### CODE REVIEW

**Comment by kgrgreer** [comment]
Why not to merge?

**Comment by noodlemoodle** [comment]
> Why not to merge?


haven't tested it very well yet, and need to fix failing tests on nanopay side

File: src/foam/dao/EasyDAO.js [hunk 1]
```diff
@@ -220,7 +220,12 @@ if ( getOrder() != null &&
   }
 }
 
-if( getAuthorize() ) delegate = new foam.nanos.auth.AuthorizationDAO(getX(), delegate, getAuthorizer());
+if( getAuthorize() ) {
```
Comment (TW80000) [inline src/foam/dao/EasyDAO.js]:
```suggestion
if ( getAuthorize() ) {
```

File: src/foam/nanos/auth/AuthorizationDAO.js [hunk 2]
```diff
@@ -57,6 +57,17 @@ foam.CLASS({
   ],
 
   methods: [
+    {
+      name: 'init_',
+      javaCode: `
+      AuthorizationException exception = new AuthorizationException("When " +
```
Comment (TW80000) [inline src/foam/nanos/auth/AuthorizationDAO.js]: Please indent the whole code block one more level.

**Review by TW80000 — CHANGES_REQUESTED** [review]

**Comment by TW80000** [comment]
As discussed in person, we're going to do the following instead:

* Make `InvalidX` extend `ProxyX`
* In `Boot.java` when initializing the services, set the context of ContextAware DAOs to `InvalidX` which delegates to the system context
* Modify `InvalidX` to log a warning rather than throw an error

**Pull request closed without merging**